### PR TITLE
Lighting Optimizations

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports some of TG's optimizations to lighting:

https://github.com/tgstation/tgstation/pull/45491
https://github.com/tgstation/tgstation/pull/43816
https://github.com/tgstation/tgstation/pull/51546

These are all minor optimizations--they won't do much, at all, to improve performance, but they will help a tiny bit---a bit less memory, a bit faster at handling things.

Also cleaned up some styling stuff

:cl: Fox McCloud
tweak: ported some improvements to the overall lighting system
/:cl:

### CODE REVIEW

File: code/modules/lighting/lighting_object.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 	icon             = LIGHTING_ICON
 	icon_state       = "transparent"
-	color            = LIGHTING_BASE_MATRIX
+	color            = null //we manually set color in init instead
```
Comment (AffectedArc07) [inline code/modules/lighting/lighting_object.dm:8]: Is this faster to set in `Initialize()` than here?

Comment (Fox-McCloud) [reply code/modules/lighting/lighting_object.dm:8]: Nah, that's not the problem.

The problem is that we have a system that creates lists of colors and their associated level of color protection on initialization. By setting the color, manually, after `. = ..()`, then we avoid thousands upon thousands of lists of color, thus cutting down on memory by a bit.

Comment (AffectedArc07) [reply code/modules/lighting/lighting_object.dm:8]: I see

File: code/controllers/subsystem/lighting.dm [hunk 2]
```diff
@@ -31,9 +30,10 @@ SUBSYSTEM_DEF(lighting)
 	MC_SPLIT_TICK_INIT(3)
 	if(!init_tick_checks)
 		MC_SPLIT_TICK
+	var/list/queue = sources_queue
 	var/i = 0
-	for(i in 1 to GLOB.lighting_update_lights.len)
-		var/datum/light_source/L = GLOB.lighting_update_lights[i]
+	for(i in 1 to length(queue))
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:35]: `LAZYLEN` here

File: code/controllers/subsystem/lighting.dm [hunk 3]
```diff
@@ -44,14 +44,15 @@ SUBSYSTEM_DEF(lighting)
...
+		queue.Cut(1, i+1)
 		i = 0
 
 	if(!init_tick_checks)
 		MC_SPLIT_TICK
 
-	for (i in 1 to GLOB.lighting_update_corners.len)
-		var/datum/lighting_corner/C = GLOB.lighting_update_corners[i]
+	queue = corners_queue
+	for(i in 1 to length(queue))
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:54]: `LAZYLEN`

File: code/controllers/subsystem/lighting.dm [hunk 4]
```diff
@@ -44,14 +44,15 @@ SUBSYSTEM_DEF(lighting)
 		else if(MC_TICK_CHECK)
 			break
 	if(i)
-		GLOB.lighting_update_lights.Cut(1, i+1)
+		queue.Cut(1, i+1)
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:47]: spacing

File: code/controllers/subsystem/lighting.dm [hunk 5]
```diff
@@ -60,15 +61,16 @@ SUBSYSTEM_DEF(lighting)
 		else if(MC_TICK_CHECK)
 			break
 	if(i)
-		GLOB.lighting_update_corners.Cut(1, i+1)
+		queue.Cut(1, i+1)
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:64]: spacing

File: code/controllers/subsystem/lighting.dm [hunk 6]
```diff
@@ -60,15 +61,16 @@ SUBSYSTEM_DEF(lighting)
...
 		i = 0
 
 
 	if(!init_tick_checks)
 		MC_SPLIT_TICK
 
-	for (i in 1 to GLOB.lighting_update_objects.len)
-		var/atom/movable/lighting_object/O = GLOB.lighting_update_objects[i]
+	queue = objects_queue
+	for(i in 1 to length(queue))
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:72]: `LAZYLEN`

File: code/controllers/subsystem/lighting.dm [hunk 7]
```diff
@@ -80,7 +82,7 @@ SUBSYSTEM_DEF(lighting)
 		else if(MC_TICK_CHECK)
 			break
 	if(i)
-		GLOB.lighting_update_objects.Cut(1, i+1)
+		queue.Cut(1, i+1)
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:85]: spacing

File: code/controllers/subsystem/lighting.dm [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
 	init_order = INIT_ORDER_LIGHTING
 	flags = SS_TICKER
 	offline_implications = "Lighting will no longer update. Shuttle call recommended."
+	var/static/list/sources_queue = list() // List of lighting sources queued for update.
+	var/static/list/corners_queue = list() // List of lighting corners queued for update.
+	var/static/list/objects_queue = list() // List of lighting objects queued for update.
 
 /datum/controller/subsystem/lighting/stat_entry()
-	..("L:[GLOB.lighting_update_lights.len]|C:[GLOB.lighting_update_corners.len]|O:[GLOB.lighting_update_objects.len]")
+	..("L:[length(sources_queue)]|C:[length(corners_queue)]|O:[length(objects_queue)]")
```
Comment (farie82) [inline code/controllers/subsystem/lighting.dm:12]: `LAZYLEN`

File: code/modules/lighting/lighting_corner.dm [hunk 9]
```diff
@@ -79,15 +79,16 @@ GLOBAL_LIST_INIT(LIGHTING_CORNER_DIAGONAL, list(NORTHEAST, SOUTHEAST, SOUTHWEST,
 	active = FALSE
 	var/turf/T
 	var/thing
-	for (thing in masters)
+	for(thing in masters)
```
Comment (farie82) [inline code/modules/lighting/lighting_corner.dm:82]: Weird syntax use here. Prob best to refactor this to use a scoped var

File: code/controllers/subsystem/lighting.dm (hunk 2 above)
Comment (Fox-McCloud) [reply code/controllers/subsystem/lighting.dm:35]: not a lazy list.

File: code/game/atoms_movable.dm [hunk 10]
```diff
@@ -265,15 +265,8 @@
 		var/dest_z = (destturf ? destturf.z : null)
 		if(old_z != dest_z)
 			onTransitZ(old_z, dest_z)
-		if(isturf(destination) && opacity)
```
Comment (Fox-McCloud) [inline code/game/atoms_movable.dm:268]: handled in `turf/Entered` now.

File: code/game/atoms_movable.dm [hunk 11]
```diff
@@ -265,15 +265,8 @@
...
-			var/turf/new_loc = destination
-			new_loc.reconsider_lights()
 
-	if(isturf(old_loc) && opacity)
```
Comment (Fox-McCloud) [inline code/game/atoms_movable.dm:272]: This was already handled in `turf/Exited`

File: code/game/atoms_movable.dm [hunk 12]
```diff
@@ -265,15 +265,8 @@
...
-		old_loc.reconsider_lights()
-
-	for(var/datum/light_source/L in light_sources)
```
Comment (Fox-McCloud) [inline code/game/atoms_movable.dm:275]: already handled in `atom/movable/Moved`

**Pull request merged**